src/module-path.ts

```diff
@@ -24,7 +24,7 @@ export function getModulePath(currentModule: string, importPath: string, host: I
     const pkgFile = host.joinPath(dir, 'package.json');
     if (host.fileExists(pkgFile) && host.isFile(pkgFile)) {
       const pkg = JSON.parse(host.readFile(pkgFile).toString());
-      const main = pkg['jsnext:main'] || (typeof pkg.browser === 'string' ? pkg.browser : undefined) || pkg.main;
+      const main = (typeof pkg.browser === 'string' ? pkg.browser : undefined) || pkg['jsnext:main'] || pkg.main;
```

Comment:
* Which falsy values are allowed in this fields?
* The `browser`-object does need a separate commit, because its a lot more work to do to have a full spec compliant implementation.